justfile

```diff
@@ -0,0 +1,62 @@
+set quiet
+
+import? '../sdk-codegen/justfile'
+
+_default:
+    just --list --unsorted
+
+# ⭐ run all unit tests, or pass a package name (./invoice) to only run those tests
+test *args="./...":
+    go run scripts/test_with_stripe_mock/main.go -race {{ args }}
+
+# check for potential mistakes (slow)
+lint: install
+    go vet ./...
+    staticcheck
+
+# ⭐ format all files
+format: install
+    scripts/gofmt.sh
+    goimports -w example/generated_examples_test.go
+
+# verify, but don't modify, the formatting of the files
+format-check:
+    scripts/gofmt.sh check
+
+# ensures all client structs are properly registered
+check-api-clients:
+    go run scripts/check_api_clients/main.go
+
+ci-test: test bench check-api-clients
+
+# compile the project
+build:
+    go build ./...
+
+# install dependencies (including those needed for development). Mostly called by other recipes
+install:
+    go get
+    go install honnef.co/go/tools/cmd/staticcheck@v0.4.7
+    go install golang.org/x/tools/cmd/goimports@v0.24.0
+
+# run benchmarking to check for performance regressions
+bench:
+    go test -race -bench . -run "Benchmark" ./form
+
+# called by tooling. It updates the package version in the `VERSION` file and `stripe.go`
+[private]
+update-version version: && _normalize-imports
+    echo "{{ version }}" > VERSION
+    perl -pi -e 's|const clientversion = "[.\d\-\w]+"|const clientversion = "{{ version }}"|' stripe.go
+
+# go imports use the package's major version in the path, so we need to update them
```

Comment:
This was the only especially tricky thing- we want `major_version` to be evaluated lazily, but also shared throughout the recipe. 

This comment is probably too detailed, but I wanted to remind myself why I did it exactly this way.